pkg/framework/plugins/nodeutilization/types.go

```diff
@@ -48,6 +48,12 @@ type HighNodeUtilizationArgs struct {
 	Thresholds    api.ResourceThresholds `json:"thresholds"`
 	NumberOfNodes int                    `json:"numberOfNodes,omitempty"`
 
+	// RequireResourceRequestForEviction makes the descheduler evict only
+	// pods that have a resource request defined for any of the user
+	// provided thresholds. If the pod does not request the resource, it
+	// will not be evicted.
+	RequireResourceRequestForEviction bool `json:"requireResourceRequestForEviction,omitempty"`
```

Comment:
One way is to limit this new option for the requested resources only. The other is to name it the way that can be easily extended in the future. E.g. when a new usage client (which makes this filter meaningful) gets introduced. What about `exclusivePodEviction` or similar name? Something giving the idea only pods with resources listed in the `thresholds` field are considered.

Reply:
Or, instead of creating a bool option what about an enum? E.g.:
```go
type EvictionMode string

var (
	EvictAll EvictionMode = "All"
	EvictOnlyThresholdingResources EvictionMode = "OnlyThresholdingResources"
)
```
. Which could be later extended with other modes as they go without deprecating the boolean option.

Reply:
Or, we could also allow to combine multiple modes:
```yaml
apiVersion: "descheduler/v1alpha2"
kind: "DeschedulerPolicy"
profiles:
  - name: ProfileName
    pluginConfig:
    - name: "HighNodeUtilization"
      args:
        thresholds:
          "cpu" : 20
          "memory": 20
          "pods": 20
        evictableNamespaces:
          exclude:
          - "kube-system"
          - "namespace1"
        evictionModes:
          - OnlyThresholdingResources
          - NoSorting
    plugins:
      balance:
        enabled:
          - "HighNodeUtilization"
```